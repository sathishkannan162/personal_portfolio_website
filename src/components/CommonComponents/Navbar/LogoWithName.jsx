import { Flex, Avatar, VisuallyHidden, chakra } from '@chakra-ui/react';
import { Link } from 'react-router-dom';

export default function LogoWithName({ Logo, Name }) {
  return (
    <Flex>
      <Link to="/">
        <Avatar src={Logo} size="sm" alt="sathish kannan logo" />
      </Link>
      <Link to="/">
        <VisuallyHidden>{Name}</VisuallyHidden>
      </Link>
      <Link to="/">
        <chakra.h1
          fontSize="xl"
          fontWeight="medium"
          ml="2"
          _hover={{ color: 'gray.600' }}
        >
          {Name}
        </chakra.h1>
      </Link>
    </Flex>
  );
}
