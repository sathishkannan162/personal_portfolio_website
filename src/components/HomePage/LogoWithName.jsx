import { Flex, Link, Avatar, VisuallyHidden, chakra } from '@chakra-ui/react';

export default function LogoWithName({ Logo, Name }) {
  return (
    <Flex>
      <Link to="/">
        <Avatar src={Logo} size="sm" />
      </Link>
      <Link to="/">
        <VisuallyHidden>{Name}</VisuallyHidden>
      </Link>
      <Link to="/">
        <chakra.h1
          fontSize="xl"
          fontWeight="medium"
          ml="2"
          _hover={{ color: 'gray.600' }}
        >
          {Name}
        </chakra.h1>
      </Link>
    </Flex>
  );
}
