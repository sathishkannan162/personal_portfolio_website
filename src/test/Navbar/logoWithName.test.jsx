import React from 'react';
import { render, screen } from '../test-utils';
import LogoWithName from '../../components/CommonComponents/Navbar/LogoWithName'

describe('LogoWithName', () => {
    it('renders the logo with name and links correctly', () => {
        const logoUrl = 'https://projectimages.netlify.app/website_logo/website_logo.png'
        const name = 'My Website';

        render(
            <LogoWithName Logo={logoUrl} Name={name} />
        );

        const logoLinks = screen.queryAllByRole('link', { name: name });
        logoLinks.forEach((logoLink) => {
            expect(logoLink).toBeInTheDocument();
            expect(logoLink).toHaveAttribute('href', '/');
        })

        const logoImage = screen.getByTestId('site-logo');
        expect(logoImage).toBeInTheDocument();
        expect(logoImage).toHaveAttribute('src', logoUrl);
        expect(logoImage).toHaveAttribute('alt', 'Sathish kannan Logo');
        expect(logoImage).toHaveStyle('width: 30px;');
        expect(logoImage).toHaveStyle('height: 30px;');

        const siteNameText = screen.getByTestId('sitename');
        expect(siteNameText).toBeInTheDocument();
        expect(siteNameText).toHaveTextContent(name);
    });
    describe('LogoWithName', () => {
        const testLogo = 'test-logo.png';
        const testName = 'Test Site';

        test('renders correctly', () => {
            const { asFragment } = render(
                    <LogoWithName Logo={testLogo} Name={testName} />
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
