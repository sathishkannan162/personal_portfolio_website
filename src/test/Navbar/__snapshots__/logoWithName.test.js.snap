// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoWithName LogoWithName renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-k008qs"
  >
    <a
      href="/"
    >
      <img
        alt="Sathish kannan Logo"
        class="chakra-image css-c2uxn6"
        data-test="site-logo"
        src="test-logo.png"
      />
    </a>
    <a
      href="/"
    >
      <span
        class="css-idkz9h"
      >
        Test Site
      </span>
    </a>
    <a
      href="/"
    >
      <h1
        class="css-1fg4y8a"
        data-test="sitename"
      >
        Test Site
      </h1>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
